mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1088,6 +1163,192 @@ func TestClientSideEncryptionProse(t *testing.T) {
+				"privateKey": gcpPrivateKey,
+				"endpoint":   "127.0.0.1:8000",
+			},
+			"kmip": {
+				"endpoint": "127.0.0.1:8000",
+			},
+		}
+
+		invalidKmsProviders := map[string]map[string]interface{}{
+			"aws": {
+				"accessKeyId":     awsAccessKeyID,
+				"secretAccessKey": awsSecretAccessKey,
+			},
+			"azure": {
+				"tenantId":                 azureTenantID,
+				"clientId":                 azureClientID,
+				"clientSecret":             azureClientSecret,
+				"identityPlatformEndpoint": "127.0.0.1:8001",
+			},
+			"gcp": {
+				"email":      gcpEmail,
+				"privateKey": gcpPrivateKey,
+				"endpoint":   "127.0.0.1:8001",
+			},
+			"kmip": {
+				"endpoint": "127.0.0.1:8001",
+			},
+		}
+
+		// create valid Client Encryption options without TLS options
+		validClientEncryptionOptionsWithoutTLS := options.ClientEncryption().
+			SetKmsProviders(validKmsProviders).
+			SetKeyVaultNamespace(kvNamespace)
+
+		// make TLS opts containing client certificate and CA file
+		tlsOptsMap := make(map[string]map[string]interface{})
+		clientAndCATlsMap := map[string]interface{}{
+			"tlsCertificateKeyFile": tlsClientCertificateKeyFile,
+			"tlsCAFile":             tlsCAFile,
+		}
+		tlsOptsMap["kmip"] = clientAndCATlsMap
+
+		// create valid Client Encryption options and set valid TLS options
+		validClientEncryptionOptionsWithTLS := options.ClientEncryption().
+			SetKmsProviders(validKmsProviders).
+			SetKeyVaultNamespace(kvNamespace)
+
+		_, err := validClientEncryptionOptionsWithTLS.SetTLSOptions(tlsOptsMap)
+		assert.Nil(mt, err, "SetTLSOptions error: %v", err)
+
+		// make TLS opts containing only CA file
+		caTlsMap := map[string]interface{}{
+			"tlsCAFile": tlsCAFile,
+		}
+		tlsOptsMap["kmip"] = caTlsMap
+
+		// create invalid Client Encryption options with expired credentials
+		expiredClientEncryptionOptions := options.ClientEncryption().
+			SetKmsProviders(expiredKmsProviders).
+			SetKeyVaultNamespace(kvNamespace)
+
+		_, err = expiredClientEncryptionOptions.SetTLSOptions(tlsOptsMap)
+		assert.Nil(mt, err, "SetTLSOptions error: %v", err)
+
+		// create invalid Client Encryption options with invalid hostnames
+		invalidHostnameClientEncryptionOptions := options.ClientEncryption().
+			SetKmsProviders(invalidKmsProviders).
+			SetKeyVaultNamespace(kvNamespace)
+
+		_, err = invalidHostnameClientEncryptionOptions.SetTLSOptions(tlsOptsMap)
+		assert.Nil(mt, err, "SetTLSOptions error: %v", err)
+
+		awsMasterKey := map[string]interface{}{
+			"region":   "us-east-1",
+			"key":      "arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0",
+			"endpoint": "127.0.0.1:8002",
+		}
+		azureMasterKey := map[string]interface{}{
+			"keyVaultEndpoint": "doesnotexist.local",
+			"keyName":          "foo",
+		}
+		gcpMasterKey := map[string]interface{}{
+			"projectId": "foo",
+			"location":  "bar",
+			"keyRing":   "baz",
+			"keyName":   "foo",
+		}
+		kmipMasterKey := map[string]interface{}{}
+
+		testCases := []struct {
```

Comment:
The `masterKey` used for AWS differs with each different ClientEncryption object used. Currently the tests uses `awsMasterKey` for all cases which has `"endpoint": "127.0.0.1:8002"`.

AWS is a special case. AWS uses a different `masterKey` for each of the `createDataKey` calls. The `masterKey` for the other providers remains constant.

I suggest defining separate masterKey variables for each of the AWS cases. Expand the test case struct to pass a masterKey for each of the `createDataKey` calls:

```suggestion
		testCases := []struct {
			name                    string
			masterKeyNoClientCert   interface{}
			masterKeyWithTLS        interface{}
			masterKeyExpired        interface{}
			masterKeyInvalidHotname interface{}
			expiredError            string
			invalidHostnameError    string
		}
```